### PR TITLE
User class branch

### LINKED ISSUE #1: Create User Model
Create User Model

### PR DESCRIPTION
**Pull request by barBen**
#1

### CODE REVIEW

**Review by ghost — CHANGES_REQUESTED** [review]
.

File: DGN/Models/User.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        [DataType(DataType.EmailAddress)]
+        public string Email { get; set; }
+
+        [Required]
+        [DisplayName("Username")]
+        public string UserName { get; set; }
+
+        [Required]
+        [RegularExpretion(@"^[A-Z]+[a-z].*$")]
+        public string Password { get; set; }
```
Comment (MaayanMordehai) [inline DGN/Models/User.cs:25]: The User should have one to one constrain with the Password :)

Comment (ghost) [inline DGN/Models/User.cs]: The User should have one to one constrain with the Password :)

**Review by ghost — CHANGES_REQUESTED** [review]

File: DGN/Models/User.cs [hunk 2]
```diff
@@ -20,6 +20,8 @@ public class User
         [DisplayName("Username")]
         public string UserName { get; set; }
 
+        public int PssswordID { get; set; }
```
Comment (ghost) [inline DGN/Models/User.cs]: need to be:
[ForeignKey("Password")]
public int PasswordId { get; set; }

**Review by ghost — CHANGES_REQUESTED** [review]

File: DGN/Models/User.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        [Required]
+        [DisplayName("Username")]
+        public string UserName { get; set; }
+
+        [ForeignKey("Password")]
+        public int PssswordID { get; set; }
+
+        [Required]
+        [RegularExpression(@"^[A-Z]+[a-z].*$")]
+        public string Password { get; set; }
```
Comment (bsuro10) [inline DGN/Models/User.cs]: Should be from class "Password"

File: DGN/Models/User.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+        [Required]
+        [DisplayName("Username")]
+        public string UserName { get; set; }
+
+        [ForeignKey("Password")]
+        public int PssswordID { get; set; }
+
+        [Required]
+        [RegularExpression(@"^[A-Z]+[a-z].*$")]
```
Comment (bsuro10) [inline DGN/Models/User.cs]: We don't regular expression, 
we just need to set up minimum characters

File: DGN/Models/User.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        public string Password { get; set; }
+
+        [DisplayName("The Profile Image Location")]
+        public string ImageLocation { get; set; }
+
+        [DisplayName("Basic info about the user")]
+        [DataType(DataType.MultilineText)]
+        public string About { get; set; }
+
+        [DisplayName("The articles that the user liked")]
```
Comment (bsuro10) [inline DGN/Models/User.cs]: We don't need it because it won't be on the form

File: DGN/Models/User.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        public IList<ArticleLikes> ArticleLikes { get; set; }
+
+        // this is one to many with articles the user wrote
+        [DisplayName("Articles The User Wrote")]
```
Comment (bsuro10) [inline DGN/Models/User.cs]: We don't need it because it won't be on the form

**Review by bsuro10 — CHANGES_REQUESTED** [review]

**Review by ghost — DISMISSED** [review]

**Review by origranot — DISMISSED** [review]
Good job!

File: DGN/Models/User.cs [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+        [DisplayName("The Profile Image Location")]
+        public string ImageLocation { get; set; }
+
+        [DisplayName("Basic info about the user")]
+        [DataType(DataType.MultilineText)]
+        public string About { get; set; }
+
+        [DisplayName("The articles that the user liked")]
+        public IList<ArticleLikes> ArticleLikes { get; set; }
```
Comment (bsuro10) [inline DGN/Models/User.cs]: The IList need to be from kind `Article`

**Review by bsuro10 — CHANGES_REQUESTED** [review]

**Review by origranot — DISMISSED** [review]
Good job!

File: DGN/Models/User.cs [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        [Required]
+        [DataType(DataType.EmailAddress)]
+        public string Email { get; set; }
+
+        [Required]
+        [DisplayName("Username")]
+        public string UserName { get; set; }
+
+        [ForeignKey("Password")]
+        public int PasswordID { get; set; }
```
Comment (bsuro10) [inline DGN/Models/User.cs]: Just to make sure write it like this: `PasswordId`

**Review by bsuro10 — CHANGES_REQUESTED** [review]

File: DGN/Models/User.cs [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+        // Primery key
+        public int Id { get; set; }
+
+        [Required]
+        [DataType(DataType.EmailAddress)]
+        public string Email { get; set; }
+
+        [Required]
+        [DisplayName("Username")]
+        public string UserName { get; set; }
```
Comment (bsuro10) [inline DGN/Models/User.cs]: You can change to `Username` and then remove [DisplayName("Username")]

**Review by bsuro10 — CHANGES_REQUESTED** [review]

**Review by ghost — DISMISSED** [review]

**Review by bsuro10 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**